### PR TITLE
Fix scanning classes delimited by tab characters

### LINKED ISSUE #15148: [v4] Tailwind bails out on candidates inside multiline JS array (Vite+Svelte setup)
## Version

`tailwind@4.0.0-beta.2` and `@tailwindcss/vite@4.0.0-beta.2`

## Environment

Output of `npx envinfo --system --binaries --browsers --npmPackages "{svelte,@sveltejs/*,vite,tailwindcss,@tailwindcss}"`:

```
  System:
    OS: Linux 6.6 Arch Linux
    CPU: (4) x64 Intel(R) Core(TM) i5-4590 CPU @ 3.30GHz
    Memory: 6.32 GB / 15.57 GB
    Container: Yes
    Shell: 3.7.1 - /usr/bin/fish
  Binaries:
    Node: 22.11.0 - ~/.volta/tools/image/node/22.11.0/bin/node
    Yarn: 4.5.1 - ~/.volta/tools/image/yarn/4.5.1/bin/yarn
    npm: 10.9.0 - ~/.volta/tools/image/node/22.11.0/bin/npm
    pnpm: 9.12.3 - ~/.volta/bin/pnpm
  Browsers:
    Chromium: 131.0.6778.69
  npmPackages:
    @sveltejs/vite-plugin-svelte: ^4.0.0 => 4.0.1
    svelte: ^5.1.3 => 5.2.7
    tailwindcss: 4.0.0-beta.2 => 4.0.0-beta.2
    vite: ^5.4.10 => 5.4.11
```

## Reproduction

https://github.com/vnphanquang/tailwind-4-svelte-candidates-in-array. Please see README for steps.

## Description

Given the following Svelte source code (in Vite context as in reproduction repo above):

```svelte
<script>
  const classes = [
    'test-red',
    'test-green',
    'test-blue',
    'test-tomato', 
  ];
</script>

{#each classes as cls}
    <div class="{cls}"></div>
{/each}
```

Assuming `test-*` has been defined using `@utility` directive in CSS. The build output does not include any of the `test-*` above.

Removing one of the `test-*` class generates the expected output. It seems here that tailwind is missing out candidates inside a JS array with length > 3 in this particular use case.

It is important to note that I did not observe such behavior in V3. I have not yet tested this specific setup with V3, however.

Thanks!

### PR DESCRIPTION
**Pull request by RobinMalfait**
This PR fixes an issue where multi-line candidates in Svelte files couldn't be found as reported in #15148

After digging in, the real culprit seems to be that the reproduction used tab `\t` characters instead of spaces and we only delimited explicitly on spaces.

Initially I couldn't reproduce this in an integration test until we (@thecrypticace and I) realised that `\t` was being used.

## Test plan:

This PR adds an integration test that fails before the fix happens. The fix itself is easy in the sense that we just use all ascii whitespace characters instead of just spaces.

Fixes: #15148

### CODE REVIEW

File: crates/oxide/src/parser.rs [hunk 1]
```diff
@@ -1542,17 +1539,45 @@ mod test {
...
+                "text-[13px]",
+                "[--my-var:1_/_2]",
+                "--my-var:1_/_2",
+                "[.foo_&]:px-[0]",
+                "[.foo_&]:[color:red]",
+            ]
+        );
+
+        let candidates = run(
+            "\n          <script>\n            const classes = [\n              'text-red-500',\n              'text-green-500',\n              'text-blue-500',\n            ]\n          </script>\n        ",
```
Comment (RobinMalfait) [inline crates/oxide/src/parser.rs:1569]: These look ugly on purpose

File: integrations/vite/svelte.test.ts [hunk 2]
```diff
@@ -252,3 +252,138 @@ test(
...
+        import tailwindcss from '@tailwindcss/vite'
+        import { svelte } from '@sveltejs/vite-plugin-svelte'
+
+        // https://vite.dev/config/
+        export default defineConfig({
+          plugins: [
+            svelte(),
+            tailwindcss(),
+            {
+              name: 'loggyMcLogFace',
```
Comment (thecrypticace) [inline integrations/vite/svelte.test.ts:286]: 🤣 do we need this anymore in the test case?

File: integrations/vite/svelte.test.ts [hunk 3]
```diff
@@ -252,3 +252,138 @@ test(
...
+        }
+      `,
+      'src/App.svelte': html`
+        <script>
+          const classes = [
+            'test-red',
+            'test-green',
+            'test-blue',
+            'test-tomato',
+            'multiple-entries-to-keep-newlines',
```
Comment (thecrypticace) [inline integrations/vite/svelte.test.ts:354]: My prettier hack is I use an empty `//` comment to do this

Comment (RobinMalfait) [reply integrations/vite/svelte.test.ts:354]: That introduced a space, so didn't trust it 😂

Comment (thecrypticace) [reply integrations/vite/svelte.test.ts:342]: Ah!

**Comment by adamwathan** [comment]
Should there be explicit tests for tabs vs. sort of sneaking it into the multi-line arrays test?

**Comment by RobinMalfait** [comment]
@adamwathan good call, I added dedicated whitespace related tests (none, spaces, tabs, and newlines) in the Rust tests. Also add an integration test for the CLI.

**Review by thecrypticace — APPROVED** [review]

**Review by philipp-spiess — APPROVED** [review]

**Comment by philipp-spiess** [comment]
@RobinMalfait Wondering if in this case, the unit tests alone are enough? The integration tests are great to know this actually fixes the issue but I'm not sure we'd really regress this without breaking the unit test also? 

I'm starting to get conservative here since more integration tests == slower CI 😬

**Comment by adamwathan** [comment]
I agree with @philipp-spiess let's just drop the integration tests here, I don't think they give me any more confidence so whatever 👍

**Comment by RobinMalfait** [comment]
Dropped them and rebased on top of `next` 👍

**Pull request merged**